app/controllers/publish/courses/applications_open_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Publish
+  module Courses
+    class ApplicationsOpenController < PublishController
+      before_action :build_recruitment_cycle
+      before_action :build_course_params, only: %i[update continue]
+      include CourseBasicDetailConcern
+
+      def update
+        super
+      end
+
+      def continue
+        super
+      end
+
+    private
+
+      def actual_params
+        params.require(:course)
+          .except(
+            :qualification,
+            :maths,
+            :english,
+            :science,
+            :funding_type,
+            :level,
+            :is_send,
+            :study_mode,
+            :age_range_in_years,
+            :sites_ids,
+            :subjects,
+            :goto_confirmation,
+            :accredited_body_code,
+          )
+          .permit(
+            :applications_open_from,
+            :day,
+            :month,
+            :year,
+          )
+      end
+
+      # This is needed to handle the fact that dates are optionally specified as year/month/day in the UI
+      # This method assigns the params to the correct YYYY-MM-DD value given what is selected
+      def build_course_params
+        if params.key?(:course)
+          applications_open_from =
+            if actual_params["applications_open_from"] == "other"
+              "#{actual_params['year']}-#{actual_params['month']}-#{actual_params['day']}"
+            else
+              actual_params["applications_open_from"]
+            end
+          params["course"]["applications_open_from"] = applications_open_from
+        else
+          ActionController::Parameters.new({}).permit
+        end
+      end
+
+      def build_recruitment_cycle
```

Comment:
We should try and standardise on the memoised selectors as per the ADR so it's all consistent - otherwise we'll just have different ways to build these things which will get confusing

Reply:
We need to have a deeper and hard look as unfortunate there is a lot of conflict of interest from 
- https://github.com/rubocop/rubocop-rails/issues/343
- I've already relaxed a whole set https://github.com/DFE-Digital/teacher-training-api/pull/2471/commits/d18238e93ae57044f5441f42aaea8acf1e2b4c77
- the usage of draper vs ADR

It is slowly becoming a choice of porting it across with minimal changes or a complete rewrite it to conform to a standard.

Reply:
No probs, let's just go with keeping things backwards compatible as much as possible. I eventually want to get rid of draper so we could leverage view components more but can do it post migration